swift/ql/src/queries/Security/CWE-312/CleartextStoragePreferences.qhelp

```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE qhelp PUBLIC "-//Semmle//qhelp//EN" "qhelp.dtd">
+<qhelp>
+<overview>
+
+<p>Sensitive information that is stored unencrypted in the defaults database is accessible to an attacker who gains access to that database. For example, the information could be accessed by any process or user in a rooted device, or exposed through another vulnerability.</p>
```

Comment:
According to https://developer.apple.com/documentation/foundation/userdefaults, preferences can also be accessed by "App extensions on macOS and iOS".  I'm not sure but it seems like this (and perhaps other aspects described in that link) widens the attack surface as well.